sdk/ml/azure-ai-ml/azure/ai/ml/distillation/_distillation.py

```diff
@@ -0,0 +1,109 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+"""Entrypoint for creating Distillation task."""
+from typing import Any, Dict, Optional, Union
+
+from azure.ai.ml.constants import DataGenerationType
+from azure.ai.ml.constants._common import AssetTypes
+from azure.ai.ml.entities._inputs_outputs import Input
+from azure.ai.ml.entities._job.distillation.distillation_job import DistillationJob
+from azure.ai.ml.entities._job.distillation.distillation_types import PromptSettings, TeacherModelSettings
+from azure.ai.ml.entities._job.resource_configuration import ResourceConfiguration
+from azure.ai.ml.entities._workspace.connections.workspace_connection import WorkspaceConnection
+
+
+def distillation(
+    experiment_name: str,
+    data_generation_type: str,
+    data_generation_task_type: str,
+    teacher_model_endpoint_connection: WorkspaceConnection,
+    student_model: Union[Input, str],
+    training_data: Optional[Union[Input, str]] = None,
+    validation_data: Optional[Union[Input, str]] = None,
+    teacher_model_settings: Optional[TeacherModelSettings] = None,
+    prompt_settings: Optional[PromptSettings] = None,
+    hyperparameters: Optional[Dict] = None,
+    resources: Optional[ResourceConfiguration] = None,
+    **kwargs: Any,
+) -> "DistillationJob":
+    """Function to create a Distillation job.
+
+    A distillation job is used to transfer knowledge from a teacher model to student model by a two step process of
+    generating synthetic data from the teacher model and then finetuning the student model with the generated
+    synthetic data.
+
+    :param experiment_name: The name of the experiment.
+    :type experiment_name: str
+    :param data_generation_type: The type of data generation to perform.
+
+        Acceptable values: data_generation, label_generation
```

Comment:
we should leave data_generation type out until we have full support

Reply:
Removed 'data_generation'